pkg/network/ovn_kubernetes.go

```diff
@@ -562,6 +562,12 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 	// when the containerized deployment is used in hypershift hosted clusters.
 	// We will rollout unless the user has rolled out its own.
 	renderIPsecMachineConfig = renderIPsecDaemonSet && !isUserIPsecMachineConfigPresent && !isHypershiftHostedCluster
+	// When IPsec MachineConfig is rendered for the first time, check machine config cluster operator is in ready state. This
+	// ensures cluster is settled with Machine Configs present during install time. If the operator is not ready, then do not
+	// render IPsec MachineConfig.
+	if renderIPsecMachineConfig && !isIPsecMachineConfigPresent(bootstrapResult.Infra) && !bootstrapResult.Infra.MachineConfigClusterOperatorReady {
```

Comment:
Why could keep the flow that we already had going on here
```
...
isMachineConfigClusterOperatorReady := bootstrapResult.Infra.MachineConfigClusterOperatorReady
isUserIPsecMachineConfigPresent := isUserIPsecMachineConfigPresent(bootstrapResult.Infra)
isCNOIPsecMachineConfigPresent := isIPsecMachineConfigPresent(bootstrapResult.Infra) && !isUserIPsecMachineConfigPresent
isIpsecMachineConfigActive := isIPsecMachineConfigActive(bootstrapResult.Infra)
isIPsecMachineConfigNotActiveOnUpgrade := isIpsecUpgrade && !isIpsecMachineConfigActive && !isHypershiftHostedCluster
...
// Wait for MCO to be ready unless we had already rendered the IPsec MachineConfig
renderIPsecMachineConfig = renderIPsecMachineConfig && (isCNOIPsecMachineConfigPresent || isMachineConfigClusterOperatorReady)
```

Probably renaming `renderIPsecMachineConfig` to `renderCNOIPsecMachineConfig` helps with the understanding

Reply:
yes, nice comment. done.